### PR TITLE
Adds support for capturing i18n translation updates within a theme

### PR DESCRIPTION
**Pull request by bbranan**
## References
_Add references/links to any related issues or PRs. These may include:_
* Supports #1299 

## Description
This PR adds a script which can be used to merge i18n translation files from within a theme with the base/default translation files. This allows translation file updates to follow the same theme pattern as most other UI updates.

Note that this PR does not add the merge script into the build or deploy process. The included script will not be run unless it is executed directly. Once the script has been tested and accepted, it may be desirable to execute the script as part of the build, pointing to the configured theme(s).

## Instructions for Reviewers
List of changes in this PR:
* Adds a script that can be run at any time to merge two i18n translation files. It is assumed that the two files to be merged will have the same name and exist in different directories.
* Updates package.json to make the script available under the name `merge-i18n`, following the pattern of the existing `sync-i18n` script.
* Adds a simple translation customization example to the custom theme. Running the script and pointing to this file (as described in the comments within the script) will result in the English translation file (en.json5) being updated such that two label values are overridden.

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes.

From within the top level /scripts directory, run `yarn merge-i18n -s src/themes/custom/assets/i18n`. The script will run and update the top-level en.json5 file in src/assets/i18n to replace the two labels included in the custom en.json5 file. Additional overrides can be added, as well as new label name/value pairs.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: scripts/merge-i18n-files.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  let parsed_source = JSON5.parse(source_file)
+  progressBar.update(30);
+  let parsed_output = JSON5.parse(output_file)
+  progressBar.update(40);
+
+  for (let key of Object.keys(parsed_source)) {
+    parsed_output[key] = parsed_source[key];
+  }
+  progressBar.update(80);
+  fs.writeFileSync(pathToOutputFile,JSON5.stringify(parsed_output,{space:'\n  '}), {encoding:'utf8'})
```
Comment (artlowel) [inline scripts/merge-i18n-files.ts:96]: If you add the `quote` option here, the output file will keep using double quotes

```suggestion
  fs.writeFileSync(pathToOutputFile,JSON5.stringify(parsed_output,{ space:'\n  ', quote: '"' }), { encoding:'utf8' });
```

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @bbranan!

The code looks good and works, I just have a few suggestions.

Currently it will replace all double quotes in the output file with single quotes. This is perfectly valid JSON5, but it will cause a lot of unnecessary changes to that file, making it hard to see what's changed in a diff view. You can retain those double quotes by adding the `quote` option to the `stringify` call.

Second, the file doesn't follow our code style rules. They aren't checked automatically because it is outside the src dir. 

If you run 

```bash
node_modules/.bin/tslint --fix scripts/merge-i18n-files.ts
```

from the root dir of the project it should fix most of those automatically, and list the issues it needs you to fix manually.

File: scripts/merge-i18n-files.ts (hunk 1 above)
Comment (bbranan) [reply scripts/merge-i18n-files.ts]: Thanks Art, the result with double quotes still pulls out duplicate newlines and unnecessary spaces, but it's certainly easier to see the changes.

**Comment by tdonohue** [comment]
@bbranan and @artlowel :  I wonder if this should be documented briefly somewhere?  It seems very useful, but only if people are aware of it.  Maybe we should minimally add a section on "i18n management" to the README.md?  

Or maybe it's better placed in the official docs here: https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Customization ? Perhaps with a note saying that the feature is coming in 7.1?

If possible, I'd appreciate help on the docs so that this feature isn't built & accidentally forgotten about. 😉

**Comment by bbranan** [comment]
@tdonohue I added documentation for this in the official docs here: https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Customization#UserInterfaceCustomization-CustomizeUIlabelsusingInternationalization(i18n)files with a note about the merge-i18n script only being available in 7.1. I didn't see a place where it fit in the README.md (or in docs/Configuration.md), so didn't make any changes there.

**Review by tdonohue — APPROVED** [review]
Thanks @bbranan !  The docs look great to me.  The changes here also look good as well.  The only (future) improvement I can think of would be a way to automate this even further (by ensuring `merge-i18n` is run as part of the build process every time, as you noted in the description).  But, that could be done in a future PR, assuming others find this useful. So, this is a great first step.

**Pull request merged**